### PR TITLE
chore: add gh CLI to image

### PR DESCRIPTION
**Pull request by ernesgonzalez33**
Needed for the create-github-release task in the release-service-catalog. To be added in this PR: https://github.com/redhat-appstudio/release-service-catalog/pull/281

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -5,12 +5,14 @@ ARG KUBECTL_VERSION=1.27.2
...
 ARG YQ_VERSION=4.34.1
 ARG GLAB_VERSION=1.31.0
+ARG GH_VERSION=2.32.1
 
 RUN curl -L https://github.com/mikefarah/yq/releases/download/v${YQ_VERSION}/yq_linux_amd64 -o /usr/bin/yq &&\
     curl -L https://dl.k8s.io/release/v${KUBECTL_VERSION}/bin/linux/amd64/kubectl -o /usr/bin/kubectl &&\
     curl -L https://mirror.openshift.com/pub/openshift-v4/x86_64/clients/ocp/${OCP_VERSION}/opm-linux.tar.gz |tar -C /usr/bin -xzf - &&\
     curl -L https://gitlab.com/gitlab-org/cli/-/releases/v${GLAB_VERSION}/downloads/glab_${GLAB_VERSION}_Linux_x86_64.tar.gz | tar -C /usr -xzf - bin/glab &&\
-    chmod +x /usr/bin/{yq,kubectl,opm,glab} &&\
+    curl -L https://github.com/cli/cli/releases/download/v2.32.1/gh_${GH_VERSION}_linux_amd64.tar.gz -o gh_${GH_VERSION}_linux_amd64.tar.gz && tar -xzf gh_${GH_VERSION}_linux_amd64.tar.gz && mv gh_${GH_VERSION}_linux_amd64/bin/gh /usr/bin/gh &&\
```
Comment (mmalina) [inline Dockerfile:14]: Why not directly pipe the tgz through tar like we do with ocp and glab? At the very least the tar file would need to be removed.

Comment (ernesgonzalez33) [reply Dockerfile:14]: Because it is a directory. So piping wasn't working. But I can remove the tar file

Comment (mmalina) [reply Dockerfile:14]: I think for glab it's also a directory. That's why we specify to only extract one file from it - `bin/glab`. Is there a reason why it wouldn't work like this in this case?

Comment (ernesgonzalez33) [reply Dockerfile]: Changed now. Adding the `strip=1` removes the top dir that was causing me problems

Comment (mmalina) [reply Dockerfile]: Cool!

File: Dockerfile [hunk 2]
```diff
@@ -5,12 +5,14 @@ ARG KUBECTL_VERSION=1.27.2
...
 ARG YQ_VERSION=4.34.1
 ARG GLAB_VERSION=1.31.0
+ARG GH_VERSION=2.32.1
 
 RUN curl -L https://github.com/mikefarah/yq/releases/download/v${YQ_VERSION}/yq_linux_amd64 -o /usr/bin/yq &&\
     curl -L https://dl.k8s.io/release/v${KUBECTL_VERSION}/bin/linux/amd64/kubectl -o /usr/bin/kubectl &&\
     curl -L https://mirror.openshift.com/pub/openshift-v4/x86_64/clients/ocp/${OCP_VERSION}/opm-linux.tar.gz |tar -C /usr/bin -xzf - &&\
     curl -L https://gitlab.com/gitlab-org/cli/-/releases/v${GLAB_VERSION}/downloads/glab_${GLAB_VERSION}_Linux_x86_64.tar.gz | tar -C /usr -xzf - bin/glab &&\
-    chmod +x /usr/bin/{yq,kubectl,opm,glab} &&\
+    curl -L https://github.com/cli/cli/releases/download/v2.32.1/gh_${GH_VERSION}_linux_amd64.tar.gz  | tar -C /usr -xzf - --strip=1 gh_${GH_VERSION}_linux_amd64/bin/gh &&\
```
Comment (mmalina) [inline Dockerfile:14]: Can we also use the variable for the `v2.32.1` dir here? Like `/v${GH_VERSION}/`?

**Review by mmalina — APPROVED** [review]

**Review by johnbieren — APPROVED** [review]

**Pull request merged**